homeassistant/components/renault/pyzeproxy.py

```diff
@@ -0,0 +1,134 @@
+"""Proxy to handle account communication with Renault servers via PyZE."""
+import asyncio
+import json
+
+from pyze.api import BasicCredentialStore, Gigya, Kamereon
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_KAMEREON_ACCOUNT_ID, LOGGER
+from .pyzevehicleproxy import PyzeVehicleProxy
+
+GIGYA_URL = "gigya_url"
+GIGYA_KEY = "gigya_key"
+KAMEREON_URL = "kamereon_url"
+KAMEREON_KEY = "kamereon_key"
+
+
+class PyzeProxy:
+    """Handle account communication with Renault servers via PyZE."""
+
+    def __init__(self, hass):
+        """Initialise proxy."""
+        LOGGER.debug("Creating PyzeProxy")
+        self.hass = hass
+        self._preload_config = None
+        self._credential_store = BasicCredentialStore()
+        self._gigya = None
+        self._kamereon = None
+        self._vehicle_links = None
+        self._vehicle_proxies = {}
+        self._vehicles_lock = asyncio.Lock()
+        self.entities = []
+
+    async def ensure_config_preloaded(self):
+        """Preload the configuration."""
+        if self._preload_config is not None:
+            return
+
+        session = aiohttp_client.async_get_clientsession(self.hass)
+
+        lng = "fr_FR"
+        url = f"https://renault-wrd-prod-1-euw1-myrapp-one.s3-eu-west-1.amazonaws.com/configuration/android/config_{lng}.json"
+
+        async with session.get(url) as response:
+            responsetext = await response.text()
+            if responsetext == "":
+                responsetext = "{}"
+            jsonresponse = json.loads(responsetext)
+
+            self._preload_config = {
+                GIGYA_KEY: jsonresponse["servers"]["gigyaProd"]["apikey"],
+                GIGYA_URL: jsonresponse["servers"]["gigyaProd"]["target"],
+                KAMEREON_KEY: jsonresponse["servers"]["wiredProd"]["apikey"],
+                KAMEREON_URL: jsonresponse["servers"]["wiredProd"]["target"],
+            }
+        self._gigya = Gigya(
+            api_key=self._preload_config[GIGYA_KEY],
+            root_url=self._preload_config[GIGYA_URL],
+            credentials=self._credential_store,
+        )
+        self._kamereon = Kamereon(
+            api_key=self._preload_config[KAMEREON_KEY],
+            root_url=self._preload_config[KAMEREON_URL],
+            credentials=self._credential_store,
+            gigya=self._gigya,
```

Comment:
This should be abstracted in a package hosted on PyPi.

Reply:
[pyze](https://github.com/jamesremuscat/pyze) could be an option 🤔

Reply:
@springstan It's already a dependency, and I'm active on that project (though not owner/maintainer).
I've added a PR there but will have to wait for agreement from @jamesremuscat
Can we use hard-coded API keys in home-assistant?

Reply:
Oh sorry did not see that you added it as a dependency. Great, that you have already added a PR 👍 


> If not, can we use hard-coded API keys in home-assistant?

I am not sure tbh, @frenck what is the best practice in such a case where an API key is the same for everyone?

Reply:
Depends a bit... why is this defined in Home Assistant as a constant? If it isn't a problem to have it public, why wasn't it added to the upstream library/package?

Reply:
FYI I have quickly compared three different locations as described [here](https://github.com/jamesremuscat/pyze#obtaining-api-keys) and have found out that the api key for `wiredProd` stays the same but for `gigyaProd` changes with every location. Some more investigation might be necessary to prove my findings. 

@epenet therefore it might be necessary to include the locale of the user when obtaining the api key instead of hardcoding it.

Reply:
@springstan, what's great about the Renault API is that it doesn't seem to matter which API key you use for Gigya.
It seems to work whether regardless of the locale specified.

Anyway, that may change in the future so I've made a tweak to the 7ea9ccc config flow to request the locale.

I've also removed the constant again, and have put the "grab from renault server" back in with a note about the remote PR.

Are you happy with that @frenck or do you want to press the PyZE package owner?

Reply:
In order to get this moving, I made a tweak to 182a602, so the user collects his own keys before starting the config flow.
Does that work for you @springstan / @frenck ?